apps/storefront/src/pages/MyOrders/index.mobile.test.tsx

```diff
@@ -0,0 +1,1238 @@
+
+      const nextPageButton = screen.getByRole('button', { name: /next page/ });
+      await userEvent.click(nextPageButton);
+
+      when(getOrders)
+        .calledWith(stringContainingAll('first: 10', 'offset: 0'))
+        .thenReturn(
+          buildGetCustomerOrdersWith({
+            data: {
+              customerOrders: {
+                totalCount: 100,
+                edges: [buildCustomerOrderNodeWith({ node: { orderId: '66996' } })],
+              },
+            },
+          }),
+        );
+
+      const previousPageButton = screen.getByRole('button', { name: /previous page/ });
+      await userEvent.click(previousPageButton);
+
+      await waitFor(() => {
+        expect(screen.getByRole('heading', { name: /66996/ })).toBeInTheDocument();
+      });
+    });
+
+    it('can search for orders', async () => {
+      const getOrders = vi.fn().mockReturnValue(buildGetCustomerOrdersWith('WHATEVER_VALUES'));
+
+      server.use(
+        graphql.query('GetCustomerOrders', ({ query }) => HttpResponse.json(getOrders(query))),
+      );
+
+      renderWithProviders(<MyOrders />, { preloadedState });
+
+      await waitForElementToBeRemoved(() => screen.queryAllByRole('progressbar'));
+
+      when(getOrders)
+        .calledWith(stringContainingAll('search: "66996"'))
+        .thenReturn(
+          buildGetCustomerOrdersWith({
+            data: {
+              customerOrders: {
+                totalCount: 1,
+                edges: [buildCustomerOrderNodeWith({ node: { orderId: '66996' } })],
+              },
+            },
+          }),
+        );
+
+      const searchBox = screen.getByPlaceholderText(/Search/);
+
+      await userEvent.type(searchBox, '66996');
+
+      await waitFor(() => {
+        expect(screen.getByRole('heading', { name: /66996/ })).toBeInTheDocument();
+      });
+    });
+
+    it('can filter orders', async () => {
+      vi.setSystemTime(new Date('21 November 2022'));
+
+      const getOrders = vi.fn().mockReturnValue(
+        buildGetCustomerOrdersWith({
+          data: {
+            customerOrders: {
+              totalCount: 1,
+              edges: [buildCustomerOrderNodeWith({ node: { status: 'Pending' } })],
+            },
+          },
+        }),
+      );
+
+      server.use(
+        graphql.query('GetCustomerOrders', ({ query }) => HttpResponse.json(getOrders(query))),
+        graphql.query('GetCustomerOrderStatuses', () =>
+          HttpResponse.json(
+            buildCustomerOrderStatusesWith({
+              data: {
+                bcOrderStatuses: [
+                  buildOrderStatusWith({ systemLabel: 'Pending', customLabel: 'Pending' }),
+                  buildOrderStatusWith('WHATEVER_VALUES'),
+                ],
+              },
+            }),
+          ),
+        ),
+      );
+
+      renderWithProviders(<MyOrders />, { preloadedState });
+
+      await waitForElementToBeRemoved(() => screen.queryAllByRole('progressbar'));
+
+      when(getOrders)
```

Comment:
positioning these just before the triggering action (clicking `apply`) makes it a little easier for readers to see why certain particular values are expected 🍹